src/main/java/pl/zz/sharefood/repository/FoodRepository.java

```diff
@@ -1,27 +1,17 @@
 package pl.zz.sharefood.repository;
 
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
 import pl.zz.sharefood.domain.Food;
 
 import java.util.List;
-import java.util.Optional;
+
 
 @Repository
 public interface FoodRepository extends JpaRepository<Food, Long> {
 
-    @Override
-    Food save(Food food);
-
-    @Override
-    List<Food> findAll();
-
-    @Override
-    Optional<Food> findById(Long id);
-
-    @Override
-    void deleteById(Long id);
+    Page<Food> findByNameIn(List<String> foodName, Pageable pageable);
```

Comment:
Nie potrzebujemy robić listy po nazwach, zwróćmy całą listę popaginowaną

Reply:
Nie rozumiem co masz na myśli

Reply:
List<String> foodName to jest niepotrzebne